src/types.jl

```diff
@@ -149,12 +507,552 @@ for Model in [:NeuralNetworkRegressor, :MultitargetNeuralNetworkRegressor]
+```julia
+X = coerce(X, :RAD=>Continuous)
+```
+Lets also make a test set:
+```julia
+(X, Xtest), (y, ytest) = partition((X, y), 0.7, multi=true);
+```
+Next, we can define a `builder`. In the following macro call, `n_in` is the number of expected input features, and rng is a RNG. `init` is the function used to generate the random initial weights of the network.
+expected input features, and rng is a RNG. `init` is the function used to generate the random initial weights of the network.
+random initial weights of the network.
+```julia
+builder = MLJFlux.@builder begin
+  init=Flux.glorot_uniform(rng)
+  Chain(Dense(n_in, 64, relu, init=init),
+        Dense(64, 32, relu, init=init),
+        Dense(32, 1, init=init))
+end
+```
+Finally, we can define the model!
+```julia
+NeuralNetworkRegressor = @load NeuralNetworkRegressor
+  model = NeuralNetworkRegressor(builder=builder,
+                                 rng=123,
+                                 epochs=20)
+```
+For our neural network, since different features likely have different scales, if we do not standardize the network may be implicitly biased towards features with higher magnitudes, or may have [saturated neurons](https://www.informit.com/articles/article.aspx%3fp=3131594&seqNum=2)  and not train well. Therefore, standardization is key!
+not standardize the network may be implicitly biased towards features with higher magnitudes, or may have [saturated neurons](https://www.informit.com/articles/article.aspx%3fp=3131594&seqNum=2)  and not train well. Therefore, standardization is key!
+magnitudes, or may have [saturated neurons](https://www.informit.com/articles/article.aspx%3fp=3131594&seqNum=2)  and not train well. Therefore, standardization is key!
+neurons](https://www.informit.com/articles/article.aspx%3fp=3131594&seqNum=2)  and not train well. Therefore, standardization is key!
+```julia
+pipe = Standardizer |> TransformedTargetModel(model, target=Standardizer)
+```
+If we fit with a high verbosity (>1), we will see the losses during training. We can also see the losses in the output of `report(mach)`
+also see the losses in the output of `report(mach)`
+```julia
+mach = machine(pipe, X, y)
+fit!(mach, verbosity=2)
+
+# first element initial loss, 2:end per epoch training losses
+report(mach).transformed_target_model_deterministic.training_losses
+
+```
+
+## Experimenting with learning rate
+
+We can visually compare how the learning rate affects the predictions:
+```julia
+plt = plot()
+
+rates = 10. .^ (-5:0)
+
+foreach(rates) do η
+  pipe.transformed_target_model_deterministic.model.optimiser.eta = η
+  fit!(mach, force=true, verbosity=0)
+  losses =
+      report(mach).transformed_target_model_deterministic.model.training_losses[3:end]
+  plot!(1:length(losses), losses, label=η)
+end
+plt #!md
+
+savefig(joinpath("assets", "learning_rate.png"))
+
+pipe.transformed_target_model_deterministic.model.optimiser.eta = 0.0001
+```
+
```

Comment:
I think we want to wrap this example up here, and drop  "Using Iteration Controls" section that follows. An in-sample performance estimate, and a comparison with the test should be plenty. Something like:

---
With the learning rate fixed we now compute a CV estimate of the performance (using all data bound to `mach`) and compare this with performance on the test set:

```julia
evaluate!(mach, resampling=CV(nfolds=5), measure=l2) 

fit!(mach) # train using new learning rate
yhat = predict(mach, Xtest)
l2(yhat, ytest)  |> mean
```